### PR TITLE
m_option: respect pathlist seperator when printing

### LINKED ISSUE #13014: Save running shaders to watch later files.
What is the accurate way to write the currently running shaders and save/write them to the watch later files?

I tried it with:
watch-later-options-append=glsl-shaders

And it does write the shaders that are currently being run but they're not formatted properly.

This is what the entry in the files look like:

`glsl-shaders=~~/shaders/Anime4K_Clamp_Highlights.glsl,~~/shaders/Anime4K_Restore_CNN_UL.glsl,~~/shaders/Anime4K_Upscale_GAN_x2_UL.glsl,~~/shaders/Anime4K_Darken_HQ.glsl`

Now when I try opening the file with the shader state saved like that, it gives me an error in the console that it cannot open the file and that the argument is invalid, now I tried manually correcting the errors in the entry in one of the watch later files by replacing the commas with colons and putting double quotations at the end of the entry and after glsl-shaders=" and that worked.

This is what the "corrected" entry was and it worked:

`glsl-shaders="~~/shaders/Anime4K_Clamp_Highlights.glsl;~~/shaders/Anime4K_Restore_CNN_UL.glsl;~~/shaders/Anime4K_Upscale_GAN_x2_UL.glsl;~~/shaders/Anime4K_Darken_HQ.glsl"`

Sooo, what's going on here? Am I saving/writing them to the file in the wrong way that they're not being saved in the proper format and thus giving an error and not loading up the next time the files are launched?

### PR DESCRIPTION
**Pull request by llyyr**
Apparently we never did this

Fixes #13014

### CODE REVIEW

**Comment by kasper93** [comment]
Why don't we support `,` separator everywhere? Or `;` for that matter should work too, no?

**Comment by llyyr** [comment]
Not sure, but we've always used `;` or `:` as separator for a list of paths, and `,` for regular string lists. I don't see the benefits of changing this tbh.

Just for documentation, https://github.com/mpv-player/mpv/commit/91583fccac85880ad2199f50463c095560791d56 forgot to make the change this PR makes. So it's been broken for a while now

**Comment by kasper93** [comment]
> I don't see the benefits of changing this behavior either way tbh.

Well, it would make config portable. Not separators are different between OS.

**Comment by guidocella** [comment]
I think it's supposed the imitate the separator of `PATH` on the respective systems. But I also don't see the advantage over always using `,` when Unix filenames can contain `:` and Windows filenames can contain `;`.

**Comment by guidocella** [comment]
I just remembered this issue https://github.com/mpv-player/mpv/issues/10450. This unnecessary difference can confuse users.

**Comment by llyyr** [comment]
I'll open another PR to collect more comments for that. This is still strictly an improvement since `opt->priv` shouldn't be ignored if provided by `OPT_PATHLIST`.

File: options/m_option.c [hunk 1]
```diff
@@ -1517,14 +1517,15 @@ static char *print_str_list(const m_option_t *opt, const void *src)
 {
     char **lst = NULL;
     char *ret = NULL;
+    const char *sep = opt->priv ? opt->priv : (char *)OPTION_LIST_SEPARATOR;
```
[the remark this answers is not in the document]
Comment (llyyr) [reply options/m_option.c]: Woops, sorry. Should be fine now unless I can't into C

File: options/m_option.c [hunk 2]
```diff
@@ -1517,14 +1517,15 @@ static char *print_str_list(const m_option_t *opt, const void *src)
 {
     char **lst = NULL;
     char *ret = NULL;
+    const char *sep = opt->priv ? opt->priv : (char[]){OPTION_LIST_SEPARATOR, 0};
```
Comment (kasper93) [inline options/m_option.c:1520]: I would suggest simpler approach. Separator should be one char only.

```suggestion
    const char sep = opt->priv ? *opt->priv : OPTION_LIST_SEPARATOR;
```

File: options/m_option.c [hunk 3]
```diff
@@ -1517,14 +1517,15 @@ static char *print_str_list(const m_option_t *opt, const void *src)
...
 
     if (!(src && VAL(src)))
         return talloc_strdup(NULL, "");
     lst = VAL(src);
 
     for (int i = 0; lst[i]; i++) {
         if (ret)
-            ret = talloc_strdup_append_buffer(ret, ",");
+            ret = talloc_strdup_append_buffer(ret, sep);
```
Comment (kasper93) [inline options/m_option.c:1528]:
```suggestion
            ret = talloc_strndup_append_buffer(ret, &sep, 1);
```

File: options/m_option.c (hunk 2 above)
Comment (llyyr) [reply options/m_option.c:1520]: `*opt->priv` would need to be `*(char *)opt->priv` but sure

**Pull request merged**